synapse/api/errors.py

```diff
@@ -348,6 +348,20 @@ def error_dict(self):
         )
 
 
+class RequestSendFailed(RuntimeError):
+    """Sending the request failed due to not being able to talk to the remote
```

Comment:
> Sending the request failed

sending which request?

I'd suggest either expanding the docstring to explain more clearly the circumstances under which this is raised, or moving the definition into `matrixfederationclient.py` so that you can see it is related to federation requests.

Reply:
I've added some more words, hopefully they should be a bit more useful. If not I'll just move it into matrixfederationclient